lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_v1task.rb

```diff
@@ -0,0 +1,596 @@
+# This class is used to manage V1 compatible tasks using the Task Scheduler V2 API
+# It is designed to be a binary compatible API to puppet/util/windows/taskscheduler.rb but
+# will only surface the features used by the Puppet scheduledtask provider
+#
+require_relative './taskscheduler2'
+require 'puppet/util/windows/taskscheduler' # Needed for the WIN32::ScheduledTask flag constants
+
+module PuppetX
+module PuppetLabs
+module ScheduledTask
+
+class TaskScheduler2V1Task
+  # The error class raised if any task scheduler specific calls fail.
+  class Error < Puppet::Util::Windows::Error; end
+
+  public
+  # Returns a new TaskScheduler object. If a work_item (and possibly the
+  # the trigger) are passed as arguments then a new work item is created and
+  # associated with that trigger, although you can still activate other tasks
+  # with the same handle.
+  #
+  # This is really just a bit of convenience. Passing arguments to the
+  # constructor is the same as calling TaskScheduler.new plus
+  # TaskScheduler#new_work_item.
+  #
+  def initialize(work_item = nil, trigger = nil)
+    @tasksched = PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2
+
+    if work_item
+      if trigger
+        raise TypeError unless trigger.is_a?(Hash)
+        new_work_item(work_item, trigger)
+      end
+    end
+  end
+
+  # Returns an array of scheduled task names.
+  #
+  # Emulates V1 tasks by appending the '.job' suffix
+  #
+  def enum
+    @tasksched.enum_task_names(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::ROOT_FOLDER,
+      include_child_folders: false,
+      include_compatibility: [PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_AT, PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V1]).map do |item|
+        @tasksched.task_name_from_task_path(item) + '.job'
+    end
+  end
+  alias :tasks :enum
+
+  # Activate the specified task.
+  #
+  def activate(task_name)
+    raise TypeError unless task_name.is_a?(String)
+    normal_task_name = normalize_task_name(task_name)
+    raise Error.new(_("Scheduled Task %{task_name} does not exist") % { task_name: normal_task_name }) unless exists?(normal_task_name)
```

Comment:
Do we have a test for this bit?

Reply:
Apparently not.

Reply:
Looks like it's already guarded in the new provider
lib/puppet/provider/scheduled_task/taskscheduler_api2.rb:32
`@task.activate(resource[:name] + '.job') if exists?`

and old provider
https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/scheduled_task/win32_taskscheduler.rb#L34

Reply:
Do we need one?

Reply:
Probably not.. I just noticed that the logic here was a little different that previous helper, especially wrt normalization.